felix/bpf-gpl/tc_preamble.c

```diff
@@ -33,7 +33,12 @@ int  cali_tc_preamble(struct __sk_buff *skb)
 	__u16 protocol = bpf_ntohs(skb->protocol);
 	/* Set the globals for the rest of the prog chain. */
 	if (protocol == ETH_P_IPV6) {
-		globals->data = __globals.v6;
+		if ((__globals.v6.jumps[PROG_INDEX_MAIN] != (__u32)-1) ||
+				(__globals.v6.jumps[PROG_INDEX_MAIN_DEBUG] != (__u32)-1)) {
+			globals->data = __globals.v6;
+		} else {
+			globals->data = __globals.v4;
+		}
```

Comment:
Could we make the 3 branches into 2? We always load `PROG_INDEX_MAIN` even if we have debug enabled, right? Idk know what the compiler would do about it but it would be good to keep the preamble code as simple/small as possible.

```suggestion
		if (protocol == ETH_P_IPV6 && (__globals.v6.jumps[PROG_INDEX_MAIN] != (__u32)-1) {
			globals->data = __globals.v6;
		} else {
			globals->data = __globals.v4;
		}
```